go/vt/vtadmin/cluster/config.go

```diff
@@ -284,6 +288,13 @@ func (cfg Config) Merge(override Config) Config {
 	return merged
 }
 
+func formatID(id string) string {
+	// gRPC can't process custom resolver names with underscores
+	id = strings.Replace(id, "_", "-", -1)
```

Comment:
Since we're here, we could address the wider issue I think just as easily. We could replace any invalid chars with dashes, or remove them as well. You can see the regex to replace any unaccepted chars with dashes here: https://go.dev/play/p/qmHmZ_KSnMI

If you wanted to do that, you probably just want to make the regexp a package variable so that we don't have to compile it every time (although I don't imagine we hit this very often so probably not a practical issue even if we did).

Reply:
Thought about this a bit more and IMO:

auto-fixing is a breaking change (what happens if I have a deployment with clusters `hello_world` and `hello-world`?), as well as its alternative of failing loudly on a "bad" name.

but the alternative is less magical, since users are having the names changed out silently beneath them, so we should maybe just do input validation and include an entry in the release notes?

Reply:
Yeah I agree that it's a breaking change - did you see my latest comment? I agree with input validation!

Reply:
Yep, that plan looks good to me!